pkg/kubectl/cmd/util/factory.go

```diff
@@ -336,6 +337,10 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			case autoscaling.GroupName:
 				return c.AutoscalingClient.RESTClient, nil
 			case batch.GroupName:
+				// XXX mapping is correct (batch/v2alpha1), the returned client is using batch/v1.
```

Comment:
@soltysh I tried to debug this.  I see that "mappingVersion" is correct, but the client seems to be overriding it. I think it happens during the "negotation" in the kubelet client factory.  Maybe @deads2k knows about that code.